packages/react-components/react-popover/src/components/PopoverSurface/usePopoverSurface.ts

```diff
@@ -78,6 +78,7 @@ export const usePopoverSurface_unstable = (
     // only close if the event happened inside the current popover
     // If using a stack of inline popovers, the user should call `stopPropagation` to avoid dismissing the entire stack
     if (e.key === 'Escape' && contentRef.current?.contains(e.target as HTMLDivElement)) {
+      e.stopPropagation();
```

Comment:
I think stopping the event propagation is always a bit of a dangerous choice, because it means users will never be able to have that event ever.

In this case, Dialog already has a check on prevent default

https://github.com/microsoft/fluentui/blob/30ff8c1f8364df94fc10c4f26c88e495cb44dd2e/packages/react-components/react-dialog/src/utils/isEscapeKeyDown.ts#L9-L14

Another way to handle this, is to do the same thing as popover, which just checks to see if the escape came from a DOM child. Since Dialogs always render in portals the inline edge case wouldn't even exist

https://github.com/microsoft/fluentui/blob/30ff8c1f8364df94fc10c4f26c88e495cb44dd2e/packages/react-components/react-popover/src/components/PopoverSurface/usePopoverSurface.ts#L77-L82

Reply:
On a longer term note we might need a better way of doing this, perhaps some kind of data-attribute would be good enough, if we checked all the parents of the event target

Reply:
The way I see preventing default should be the way to go for now, and then maybe we should create a task to investigate a more "definitive" solution.